extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosExporter.java

```diff
@@ -167,38 +174,125 @@ public GooglePhotosExporter(GoogleCredentialFactory credentialFactory, JsonFacto
     }
     ContinuationData continuationData = new ContinuationData(nextPageData);
 
-    List<PhotoModel> photos = new ArrayList<>();
-    for (GoogleMediaItem mediaItem : mediaItemSearchResponse.getMediaItems()) {
-      if (mediaItem.getMediaMetadata().getPhoto() != null) {
-        // TODO: address videos later on
-        photos.add(new PhotoModel(
-            "", // TODO: no title?
-            mediaItem.getBaseUrl() + "=d",
-            mediaItem.getDescription(),
-            mediaItem.getMimeType(),
-            mediaItem.getId(),
-            albumId,
-            false));
-      }
+    PhotosContainerResource containerResource = null;
+    GoogleMediaItem[] mediaItems = mediaItemSearchResponse.getMediaItems();
+    if (mediaItems != null && mediaItems.length > 0) {
+      List<PhotoModel> photos = convertPhotosList(albumId, mediaItems, jobId);
+      containerResource = new PhotosContainerResource(null, photos);
     }
 
-    PhotosContainerResource containerResource = new PhotosContainerResource(null, photos);
-
     ResultType resultType = ResultType.CONTINUE;
     if (nextPageData == null) {
       resultType = ResultType.END;
     }
+
     return new ExportResult<>(resultType, containerResource, continuationData);
   }
 
+
+  @VisibleForTesting
+  void populateContainedPhotosList(UUID jobId, TokensAndUrlAuthData authData)
+      throws IOException {
+    // Get list of all album ids
+    List<String> albumIds = new LinkedList<>();
```

Comment:
Any reason to have albumIds as a separate variable, this seems like just two nested loops is a little cleaner right?